allennlp/modules/transformer/self_attention.py

```diff
@@ -72,6 +72,17 @@ def _transpose_for_scores(self, x):
         x = x.view(*new_x_shape)
         return x.permute(0, 2, 1, 3)
 
+    @staticmethod
+    def _apply_mask(values: torch.FloatTensor, mask: torch.BoolTensor) -> torch.FloatTensor:
+        if len(mask.shape) == 2:
+            # We create a 4D attention mask from a 2D tensor mask.
+            # The shape is `batch_size x 1 x 1 x seq_len` which is broadcast
+            # to `batch_size x num_attention_heads x seq_len x seq_len`
+            mask = mask.unsqueeze(1).unsqueeze(2)
+        # `mask==1` to convert float tensors.
+        mask = (~(mask == 1)) * -10e5  # to ensure that the model also works in half-precision mode.
+        return values + mask
```

Comment:
If we need this in many places, can we put it into a single place? This is always the same for all transformers, right? So some general place for the toolkit?

Reply:
Done.